plugin/src/main/java/io/fabric8/maven/plugin/ResourceMojo.java

```diff
@@ -281,7 +302,121 @@ private boolean isOpenShiftMode() {
+        if (template != null) {
+            PodSpec podSpec = template.getSpec();
+            if (podSpec != null) {
+                Map<String, String> nameToMount = new TreeMap<>();
+                List<Volume> volumes = podSpec.getVolumes();
+                if (volumes != null) {
+                    for (Volume volume : volumes) {
+                        PersistentVolumeClaimVolumeSource persistentVolumeClaim = volume.getPersistentVolumeClaim();
+                        if (persistentVolumeClaim != null) {
+                            String name = volume.getName();
+                            nameToMount.put(name, "");
+                        }
+                    }
+                }
+
+                if (!nameToMount.isEmpty()) {
+                    List<Container> containers = podSpec.getContainers();
+                    if (containers != null) {
+                        for (Container container : containers) {
+                            List<VolumeMount> volumeMounts = container.getVolumeMounts();
+                            if (volumeMounts != null) {
+                                for (VolumeMount volumeMount : volumeMounts) {
+                                    String name = volumeMount.getName();
+                                    String mountPath = volumeMount.getMountPath();
+                                    String current = nameToMount.get(name);
+                                    if (current != null && current.length() == 0) {
+                                        nameToMount.put(name, mountPath);
+                                    }
+                                }
+                            }
+                        }
+                    }
+
+                    StringBuffer mountPaths = new StringBuffer();
+                    StringBuffer mounts = new StringBuffer();
+                    int idx = 0;
+                    for (Map.Entry<String, String> entry : nameToMount.entrySet()) {
+                        String name = entry.getKey();
+                        String mountPath = entry.getValue();
+                        mountPaths.append(", \"" + mountPath + "\"");
+
+                        String separator = ++idx >= nameToMount.size() ? "" : ",";
+                        mounts.append("                    {\n" +
+                                "                        \"name\": \"" + name + "\",\n" +
+                                "                        \"mountPath\": \"" + mountPath + "\"\n" +
+                                "                    }" + separator + "\n");
+                    }
+
+                    String pvAnnotation = "[\n" +
+                            "            {\n" +
+                            "                \"name\": \"init\",\n" +
+                            "                \"image\": \"busybox\",\n" +
+                            "                \"command\": [\"chmod\", \"777\"" + mountPaths + "],\n" +
```

Comment:
As mentioned previously, don't really understand why this needed. Any reference to an issue or so ?